### PR TITLE
Fixed for iotagent-json issue #426

### PR DESCRIPTION
**Pull request by DeepshikhaNEC**
Fixed issue #426 for "Log always writes the same correlator and transaction id".

### CODE REVIEW

File: lib/commonBindings.js [hunk 1]
```diff
@@ -303,6 +305,7 @@ function amqpMessageHandler(topic, message) {
  * @param {Object} message      MQTT message body (Object or Buffer, depending on the value).
  */
 function mqttMessageHandler(topic, message) {
+    regenerateTransid(topic);
```
Comment (fgalan) [inline lib/commonBindings.js:308]: For the sake of homogeneity, I think it would be a good idea to apply the same solution also to the HTTP tranport. That way, we will be using the same solution for all the transports supported by this agent (HTTP, MQTT and AMQP).

Don't worry about my PR https://github.com/telefonicaid/iotagent-ul/pull/327. That PR was old and only a draft. Your solution looks smarter, so I can close my PR at the end when yours get completed.

Thanks!

Comment (DeepshikhaNEC) [reply lib/commonBindings.js:308]: > For the sake of homogeneity, I think it would be a good idea to apply the same solution also to the HTTP tranport. That way, we will be using the same solution for all the transports supported by this agent (HTTP, MQTT and AMQP).
> 
> Don't worry about my PR [telefonicaid/iotagent-ul#327](https://github.com/telefonicaid/iotagent-ul/pull/327). That PR was old and only a draft. Your solution looks smarter, so I can close my PR at the end when yours get completed.
> 
> Thanks!

@fgalan  Hi, I have checked the code for HTTP as well and it is already working fine. The functionality for MQTT and AMQP has been added in a way that they are maintaining the homogeneity.
However the code flow of http is different from the MQTT and AMQP protocol, So if it is required to sync the HTTP code as well, then there would be a huge amount of code will be changed. I suggest to keep HTTP code same as implemented by you in PR telefonicaid/iotagent-ul#327.
Please let me know your opinion.

Comment (fgalan) [reply lib/commonBindings.js:308]: So,

* The HTTP code is working well (with regards to correlatorid) now (as in master branch, I mean) in IOTA-JSON but *not* in IOTA-UL
* The HTTP code will work well (with regards to correlatorid) in IOTA-UL when https://github.com/telefonicaid/iotagent-ul/pull/327 gets merged

Is my understanding correct?

Comment (DeepshikhaNEC) [reply lib/commonBindings.js:308]: > So,
> 
> * The HTTP code is working well (with regards to correlatorid) now (as in master branch, I mean) in IOTA-JSON but _not_ in IOTA-UL
> * The HTTP code will work well (with regards to correlatorid) in IOTA-UL when [telefonicaid/iotagent-ul#327](https://github.com/telefonicaid/iotagent-ul/pull/327) gets merged
> 
> Is my understanding correct?

@fgalan Hi, The HTTP code is working well (with regards to correlatorid) for both  IOTA-JSON and IOTA-UL even without merging you PR telefonicaid/iotagent-ul#327.

Please find the attached screenshots(first one is for IOTA-UL and second one is for IOTA-JSON) according to which the new transaction id/ correlator id is generating uniquely after every new http request for both iotagent-json and iotagent-ul.

I suggest we can keep the same code (as in master branch) in both for http request as it is already working fine. 

And for MQTT and AMQP I have been raised my PR regards the iotagent-json issue #426.

totagent-ul-afterhttpreq-ss
iotagent-json-afterhttpreq-ss

File: package.json [hunk 2]
```diff
@@ -60,7 +60,7 @@
     "body-parser": "1.18.3",
     "dateformat": "3.0.3",
     "express": "~4.16.4",
-    "iotagent-node-lib": "git://github.com/telefonicaid/iotagent-node-lib.git#master",
+    "iotagent-node-lib": "git://github.com/DeepshikhaNEC/iotagent-node-lib.git#nodelib-issue/426",
```
Comment (fgalan) [inline package.json:63]: PR https://github.com/telefonicaid/iotagent-node-lib/pull/864 has been just merged into master, so please revert this change.

File: lib/commonBindings.js (hunk 1 above)
Comment (fgalan) [reply lib/commonBindings.js:308]: > I suggest we can keep the same code (as in master branch) in both for http request as it is already working fine.

Great! We would follow your suggestion. My old PR goes to "on hold" state (and in the future it will be closed).

Thanks for the analysis work!

NTC

File: package.json (hunk 2 above)
Comment (DeepshikhaNEC) [reply package.json]: > PR [telefonicaid/iotagent-node-lib#864](https://github.com/telefonicaid/iotagent-node-lib/pull/864) has been just merged into master, so please revert this change.

@fgalan  Hi, I have been reverted the changes of package.json
